src/main/java/com/google/firebase/FirebaseOptions.java

```diff
@@ -71,6 +76,26 @@ FirebaseCredential getCredential() {
     return databaseAuthVariableOverride;
   }
 
+  /**
+   * Returns the HttpTransport used to call remote HTTP endpoints.
+   *
+   * @return A Google API client HttpTransport instance.
+   */
+  @NonNull
+  public HttpTransport getHttpTransport() {
+    return httpTransport;
+  }
+
+  /**
+   * Returns the JsonFactory used to parse JSON when calling remote HTTP endpoints.
+   *
+   * @return A Google API client JsonFactory instance.
+   */
+  @NonNull
+  public JsonFactory getJsonFactory() {
+    return jsonFactory;
+  }
+
   @Override
   public boolean equals(Object obj) {
     if (!(obj instanceof FirebaseOptions)) {
```

Comment:
I will do this in a separate PR. The existing implementations in `FirebaseOptions` are ok for now.